include/JS/student_profile.js

```diff
@@ -0,0 +1,139 @@
+
+// ********************************************************************************************************************************
+//Save notes
+// ********************************************************************************************************************************
+
+function showSaveButton(heading){   
+    document.getElementById(heading).style.visibility = 'visible';
+}
+
+// ********************************************************************************************************************************
+//Saving the notes using AJAX seems like a good idea to me since I want that to be
+//saved without reloading the page
+// ********************************************************************************************************************************
+
+function savePrivateNotes(studentId){
+
+    let note = {       
+        StudentId: studentId,
+        Notes: document.getElementById('privateNotesTextarea').value
+    };
+
+    let notes = new FormData();
+    notes.append( "SavePrivateNotes", JSON.stringify(note));     
+
+    //So here I'm trusting that the DB was updated, in the future I could return a flag from
```

Comment:
This is exactly how we do it in our app -- we call it an "optimistic write". We trust that the client creates the data correctly, send it off to the server, and then check to see that the server approved it. This makes the app feel a lot faster since you aren't waiting to change the UI until the server responds.